Fix rock for MLflow 1.25

misohu opened the pull request:
Closes: https://github.com/canonical/base-mlflow/issues/15 

We are moving to use rock only instead of the Docker image. I have set the secrets to point to correct docker user. 

**Note:** I am installing MLFLow in Rock directly rather then installing it from requirements as that was the way we used in the Dockerfile.

This rock has been also tested from my personal dockerhub account directly in mlflow operator in this [PR ](https://github.com/canonical/mlflow-operator/pull/255) (so we know its working).

kimwnasptd commented:
Could we also update the `README.md` of this repo since we are bumping the version?

orfeas-k commented on tests/__pycache__/test_rock.cpython-38-pytest-8.3.2.pyc:
We don't need to commit this. We could remove this and also add a [gitignore file](https://github.com/canonical/kubeflow-rocks/blob/main/.gitignore)

orfeas-k commented on .github/workflows/on_pull_request.yaml:
Should we test with the latest supported (1.29)?

kimwnasptd commented on README.md:
```diff
@@ -1,39 +1,39 @@
-# Ubuntu 22.04 based MLflow image v2.1.1
```
Could we also add the badges for the CI, similarly to 
https://github.com/canonical/training-operator-rock/blob/main/README.md

orfeas-k commented on README.md:
````diff
@@ -1,39 +1,39 @@
-To build ROCK image manually:
-```
+To build the ROCK image manually:
+```bash
 rockcraft pack
 ```
 
-To copy resulting image `base-mflow_v2.1.1_amd64.rock` to Docker:
+To copy the resulting image `base-mlflow_v2.15.0_amd64.rock` to Docker:
+```bash
+sudo skopeo --insecure-policy copy oci-archive:base-mlflow_v2.15.0_amd64.rock docker-daemon:base-mlflow_v2.15.0_amd64.rock:rock
 ```
````
Should we replace those commands with the new tox environments? This way, we won't need to maintain those too.

orfeas-k commented on README.md:
````diff
@@ -1,39 +1,39 @@
+You can use this image seamlessly as any other Python Docker image by specifying it as a base with:
+
+```dockerfile
+FROM charmedkubeflow/mlflow:latest
````
Is the image name also changing? Won't it be `base-mlflow` anymore?

misohu replied:
sorry it always was mlflow ... I fixed it ... just the repo is called base-mlflow

orfeas-k commented on README.md:
```diff
@@ -1,39 +1,39 @@
-The image is currently published [here](https://hub.docker.com/r/charmed/base-mlflow)
-You can use this image seamlesly as any other python Docker image by specifying it as a base with.
+# Ubuntu 22.04 based MLflow image v2.15.0
+This repository regularly builds a new [MLflow](https://github.com/mlflow/mlflow) image based on LTS Ubuntu 22.04. After the new image is built, the CI runs simple tests against the image and runs the [trivy](https://github.com/aquasecurity/trivy) vulnerability scan. 
```
I understand that the part about the CI is not entirely true. And also, it's not clear to me when what this happens. I mean if we refer to the PR CI, then this builds, scans, tests and also publishes the image.

orfeas-k reviewed (commented):
Good job @misohu! I have the following questions:
* Do we use `mlflow.yaml`? If yes, right now it looks out of date. If not, we can remove it.
* In a daily ([13 feb 2024](https://docs.google.com/document/d/1WYWf8BEbB_nPasVwBWaDhhmnu461hZdFNVSAA4an1-s/edit#heading=h.n0uyh4gpbq41)), we had decided to use nested folders for single-rock repositories, thus I think we should refactor this one as well (directory name should be the same as the rock name). WDYT?

orfeas-k reviewed (approved):
Good job @misohu :)

kimwnasptd commented:
Do we still need the Dockerfile now that we converted to `rockcraft.yaml` file?

misohu commented:
Hey @kimwnasptd @orfeas-k, I have removed the Docker image dependancies ... because of that I could move the mlflow to requirements. I have pinned requirements and added update-requirements tox. Now I am testing the image in the charm here https://github.com/canonical/mlflow-operator/actions/runs/10301193187/job/28512189629

orfeas-k reviewed (approved):
Good job once more!

kimwnasptd reviewed (approved):
Looks good Michal!
